frontend/src/lib/components/TaxonomicFilter/infiniteListLogic.ts

```diff
@@ -110,6 +117,28 @@ export const infiniteListLogic = kea<infiniteListLogicType>({
                         response = apiCache[url]
                     } else {
                         response = await api.get(url)
+
+                        if (values.group?.type === TaxonomicFilterGroupType.EventProperties) {
+                            let propertiesToAdd: PropertyDefinition[] = []
+
+                            const resultsWithExcludedPropertiesRemoved = (response?.results || []).filter(
+                                (r: PropertyDefinition) => !excludedProperties.includes(r.name)
+                            )
+
+                            if (offset === 0) {
+                                // TODO this filtering should be moved to the backend and/or hidden behind the solution to https://github.com/PostHog/posthog/issues/8257
+                                propertiesToAdd = [...reservedProperties]
+                                if (searchQuery.length > 0) {
+                                    propertiesToAdd = propertiesToAdd.filter((p) => p.name.indexOf(searchQuery) >= 0)
+                                }
+                            }
+                            response = {
+                                ...response,
+                                count: propertiesToAdd.length + (resultsWithExcludedPropertiesRemoved.length ?? 0),
+                                results: [...propertiesToAdd, ...resultsWithExcludedPropertiesRemoved],
+                            }
+                        }
+
```

Comment:
This very specific code to patch only `TaxonomicFilterGroupType.EventProperties`'s infinite list results doesn't really belong in this generic component.

I propose to do all of this inside `api/property_definition.py`. It even [already supports [hiding property definitions](https://github.com/PostHog/posthog/blob/bdfe09a06b3ae5041cfcc48b864af6ebaf7cad33/posthog/api/property_definition.py#L13-L28)).

Reply:
I thought it would be a pain to duplicate the searching and filtering the SQL was doing but came up with a way to keep that in the SQL itself